clustered-indexes.md

```diff
@@ -1,188 +1,225 @@
 
-+-------------+---------+------+--------------------------------------------------------------------+---------------+
-| id          | estRows | task | access object                                                      | operator info |
-+-------------+---------+------+--------------------------------------------------------------------+---------------+
-| Point_Get_1 | 1.00    | root | table:composite_primary_key, clustered index:PRIMARY(key_a, key_b) |               |
-+-------------+---------+------+--------------------------------------------------------------------+---------------+
-1 row in set (0.00 sec)
-```
+The clustered indexes feature is partially supported in TiDB v3.0 and v4.0. It is enabled by default when the following requirements are fully met:
+
+- The table contains a `PRIMARY KEY`.
+- The `PRIMARY KEY` consists of only one column.
+- The `PRIMARY KEY` is an `INTEGER`.
 
-This behavior is consistent with MySQL, where the InnoDB storage engine will by default cluster by any `PRIMARY KEY`.
+However, since v5.0, TiDB creates all types of primary keys as non-clustered indexes by default. Because this behavior change may cause TiDB in the default configuration to perform worse in some scenarios, you can consider explicitly specifying clustered indexes.
 
-## Storage considerations
+### Compatibility with MySQL
 
-Because the `PRIMARY KEY` replaces a 64-bit `handle` value as the internal pointer to table rows, using clustered indexes might increase storage requirements. This is particularly impactful on tables that contain many secondary indexes. Consider the following example:
+TiDB specific comment syntax supports wrapping the keywords `CLUSTERED` and `NONCLUSTERED` in a comment. The result of `SHOW CREATE TABLE` also contains TiDB specific SQL comments. MySQL databases and TiDB databases of a lower version can identify and execute DDL statements with these comments.
+
+### Compatibility with TiDB ecosystem tools
+
+The clustered indexes feature is only compatible with the following ecosystem tools in v5.0 and later versions:
+
+- Backup and recovery tools: BR, Dumpling, and TiDB Lightning.
+- Data migration and replication tools: DM and TiCDC.
+
+However, you cannot convert a table with non-clustered indexes to a table with clustered indexes by using the v5.0 BR to backup and recover tables, and vice versa.
+
+### Compatibility with other TiDB features
+
+For a table with a combined primary key or a single non-integer primary key, if you change the primary key from a non-clustered index to a clustered index, the keys of its row data change as well. Therefore, `SPLIT TABLE BY/BETWEEN` statements that are executable on TiDB v5.0 and lower versions, are no longer workable in v5.0 and higher versions of TiDB. If you want to split a table with clustered indexes using `SPLIT TABLE BY/BETWEEN`, you need to provide the value of the primary key column, instead of specifying an integer value. See the following example:
```

Comment:
```suggestion
For a table with a combined primary key or a single non-integer primary key, if you change the primary key from a non-clustered index to a clustered index, the keys of its row data change as well. Therefore, `SPLIT TABLE BY/BETWEEN` statements that are executable in TiDB v5.0 and earlier versions are no longer workable in v5.0 and later versions of TiDB. If you want to split a table with clustered indexes using `SPLIT TABLE BY/BETWEEN`, you need to provide the value of the primary key column, instead of specifying an integer value. See the following example:
```

Reply:
`SPLIT TABLE BY/BETWEEN` statements that are executable in **TiDB v5.0 and earlier versions** are no longer workable in v5.0 and later versions of TiDB.

Please notice that it does not include v5.0 itself.

`SPLIT TABLE BY/BETWEEN` statements that are executable in **TiDB versions earlier than v5.0** are no longer work in v5.0 and later versions of TiDB.

Reply:
```suggestion
For a table with a combined primary key or a single non-integer primary key, if you change the primary key from a non-clustered index to a clustered index, the keys of its row data change as well. Therefore, `SPLIT TABLE BY/BETWEEN` statements that are executable in TiDB versions earlier than v5.0 are no longer workable in v5.0 and later versions of TiDB. If you want to split a table with clustered indexes using `SPLIT TABLE BY/BETWEEN`, you need to provide the value of the primary key column, instead of specifying an integer value. See the following example:
```